src/Table/Table.component.tsx

```diff
@@ -0,0 +1,266 @@
+
+interface TableHeadProps
+    extends React.HTMLAttributes<HTMLTableSectionElement> {}
+
+export const TableHead = ({
+    className,
+    children,
+    ...props
+}: TableHeadProps): React.ReactElement<TableHeadProps> => (
+    <thead className={classNames('anchor-table-head', className)} {...props}>
+        {children}
+    </thead>
+);
+
+Table.Head = TableHead;
+
+// Table.Footer
+// ----------------------------------------------------------------------------------
+
+interface TableFooterProps
+    extends React.HTMLAttributes<HTMLTableSectionElement> {}
+
+export const TableFooter = ({
+    className,
+    children,
+    ...props
+}: TableFooterProps): React.ReactElement<TableFooterProps> => (
+    <tfoot className={classNames('anchor-table-footer', className)} {...props}>
+        {children}
+    </tfoot>
+);
+
+Table.Footer = TableFooter;
+
+// Table.Cell
+// ----------------------------------------------------------------------------------
+
+interface TableCellProps
+    extends React.HTMLAttributes<HTMLTableCellElement>,
+        FontSizeProps,
+        FontWeightProps,
+        TextAlignProps,
+        WidthProps,
+        MaxWidthProps,
+        MinWidthProps,
+        PaddingProps,
+        HeightProps {
+    align?: 'center' | 'left' | 'right';
+}
+
+const StyledTd = styled('td')<TableCellProps>`
+    ${heightStyles}
+    ${widthStyles}
+    ${maxWidthStyles}
+    ${minWidthStyles}
+    ${paddingStyles}
+    ${fontSizeStyles}
+    ${fontWeightStyles}
+    ${colorStyles}
+    ${({ align }) => textAlignStyles({ textAlign: align })}
+`;
+
+export const TableCell = ({
+    className,
+    children,
+    ...props
+}: TableCellProps): React.ReactElement<TableCellProps> => (
+    <StyledTd className={classNames('anchor-table-cell', className)} {...props}>
+        {children}
+    </StyledTd>
+);
+
+Table.Cell = TableCell;
+
+// Table.HeaderCell
+// ----------------------------------------------------------------------------------
+
+interface TableHeaderCellProps
+    extends React.HTMLAttributes<HTMLTableHeaderCellElement>,
+        FontSizeProps,
+        FontWeightProps,
+        TextAlignProps,
+        WidthProps,
+        MaxWidthProps,
+        MinWidthProps,
+        PaddingProps,
+        HeightProps {
+    align?: 'center' | 'left' | 'right';
+    scope?: string;
+}
+
+const StyledTh = styled('th')<TableHeaderCellProps>`
+    ${heightStyles}
+    ${widthStyles}
+    ${maxWidthStyles}
+    ${minWidthStyles}
+    ${paddingStyles}
+    ${fontSizeStyles}
+    ${fontWeightStyles}
+    ${colorStyles}
+    ${({ align }) => textAlignStyles({ textAlign: align })}
+`;
+
+export const TableHeaderCell = ({
+    className,
+    children,
+    ...props
+}: TableHeaderCellProps): React.ReactElement<TableHeaderCellProps> => (
+    <StyledTh
+        className={classNames('anchor-table-header-cell', className)}
+        {...props}
+    >
+        {children}
+    </StyledTh>
+);
+
+Table.HeaderCell = TableHeaderCell;
```

Comment:
Nice choice on these exports. I typically try to have them all in independent files, but given our chats about this component, this really sets us up perfectly to scale what we need to and throw away what we don't without a bunch of useless boilerplate code.

Reply:
Somewhat related; perhaps the `Cell` in `Grid` should follow this pattern to be consistent. I don't generally like name-spacing the components but `Cell` for `Table` and `Cell` for `Grid`, it's having me rethink my position.

Reply:
This is a hill I won't die on. I like the dedication to consistency and I'm happy to go either way for the sake of what I'll dub "strict consistency" or "moderate consistency."

Reply:
I think the grid Cell is more fundamental, and I'm fine with it not being namespaced.